### ISSUE TITLE
You cannot access this page directly

### ISSUE THREAD

**Issue by mastef** [hybridauth/hybridauth#534]
Was getting an increased amount of 404s from LinkedIn with this thrown error. Not sure if it's really provider related, however what we found is that when Safari is configured to accept cookies "from current website only", then the session seems to be lost after the LinkedIn redirect somewhere along the way.

Because the user is assigned a new session, he of course doesn't have the config available that was saved in the session storage.

The only workaround that we found currently, is by attaching the PHPSESSID in redirects and in the endpoints, and then adding a condition for session_start if $_REQUEST["PHPSESSID"] was present, to take over that session.

A little bit less secure, because it allows for session hijacking, but works for cases where browsers simply don't save cookies because of maybe protocol or host mismatches.

So suggestion is of course to limit the session parameter based on e.g. IP or time, or adding some kind of other limitations for it. For us it works, since we just trigger a microservice queue resolution, and the session in itself is useless without the proper call + code from the oauth provider.

TL;DR : Safari can be configured to be stricter with cookies, that's when you might encounter this error. In that case you can't rely on the cookie, so you have to pass on the session var in each redirect URL, and then call session_start with it if present.

**Comment by malle-pietje** [comment]
Hi mastef, I was wondering how you add the PHPSESSID to the endpoint? Just as a GET parameter or with url rewriting?

**Comment by cirofeitosa** [comment]
Same problem here. 
Per day, 25 users have problem in login via Facebook or Google using Hybridauth.
I've tried many ways to fix, but not successful.

**Comment by StorytellerCZ** [comment]
Closing as this is a duplicate of #336.
